src/backend/core/api/viewsets.py

```diff
@@ -184,12 +176,69 @@ class UserViewSet(
     mixins.UpdateModelMixin,
     viewsets.GenericViewSet,
 ):
-    """User ViewSet"""
+    """
+    User ViewSet for all interactions with user infos and teams.
+
+    GET /api/users/&q=query
+        Return list of all users whose email matches the query, completely or partially
+    """
 
     permission_classes = [permissions.IsSelf]
     queryset = models.User.objects.all().select_related("profile_contact")
     serializer_class = serializers.UserSerializer
 
+    def list(self, request, *args, **kwargs):
+        """Limit listed users by a query with throttle protection."""
+        user = self.request.user
+        queryset = self.filter_queryset(self.get_queryset())
+
+        if not user.is_authenticated:
+            return queryset.none()
+
+        # Exclude contacts that:
+        queryset = queryset.filter(
+            # - inactive users
+            is_active=True,
+        )
+
+        # Search by case-insensitive and accent-insensitive trigram similarity
+        if query := self.request.GET.get("q", ""):
+            query = Func(Value(query), function="unaccent")
+            similarity = TrigramSimilarity(
+                Func("email", function="unaccent"),
+                query,
+            )
```

Comment:
Shouldn't we allow searching by `full_name`, `short_name` too, as for contacts ?

Reply:
I would need them though:
![image](https://github.com/numerique-gouv/people/assets/25994652/4eebac38-b1db-4c6d-a3db-68032087a479)

Reply:
Hm, it feels like we'll have to store them finally. We'll have to discuss that point with Samuel